src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudWindowsIT.java

```diff
@@ -0,0 +1,299 @@
+
+        ServiceAccountConfig sac = new StringJsonServiceAccountConfig(serviceAccountKeyJson);
+        Credentials c = (Credentials) new GoogleRobotPrivateKeyCredentials(projectId, sac, null);
+
+        CredentialsStore store = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins);
+        store.addCredentials(Domain.global(), c);
+
+        // Add Cloud plugin
+        ComputeEngineCloud gcp = new ComputeEngineCloud(CLOUD_NAME, projectId, projectId, "10", null);
+
+        // Capture log output to make sense of most failures
+        cloudLogger = LogManager.getLogManager().getLogger("com.google.jenkins.plugins.computeengine.ComputeEngineCloud");
+        if (cloudLogger != null)
+            cloudLogger.addHandler(sh);
+
+        assertEquals(0, r.jenkins.clouds.size());
+        r.jenkins.clouds.add(gcp);
+        assertEquals(1, r.jenkins.clouds.size());
+
+        // Get a compute client for out-of-band calls to GCE
+        ClientFactory clientFactory = new ClientFactory(r.jenkins, new ArrayList<DomainRequirement>(), projectId);
+        client = clientFactory.compute();
+        assertNotNull("ComputeClient can not be null", client);
+
+        // Other logging
+        clientLogger = LogManager.getLogManager().getLogger("com.google.jenkins.plugins.computeengine.ComputeClient");
+        if (clientLogger != null)
+            clientLogger.addHandler(sh);
+
+        deleteIntegrationInstances(true);
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        log.info("teardown");
+        deleteIntegrationInstances(false);
+        sh.close();
+        log.info(logOutput.toString());
+    }
+
+    @After
+    public void after() {
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.configurations.clear();
+    }
+
+    @Test
+    public void testGoogleCredentialsCreated() {
+        List<Credentials> creds = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins).getCredentials(Domain.global());
+        assertEquals(1, creds.size());
+    }
+
+    @Test //TODO: Group client tests into their own test class
+    public void testGetImage() throws Exception {
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        Image i = cloud.client.getImage(this.bootDiskProjectId, this.bootDiskImageName);
+        assertNotNull(i);
+    }
+
+    @Test(timeout = 300000)
+    public void testWorkerCreated() throws Exception {
+        //TODO: each test method should probably have its own handler.
+        logOutput.reset();
+
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfiguration1());
+
+        // Test our private key credentials were created properly
+        List<Credentials> creds = new SystemCredentialsProvider.ProviderImpl().getStore(r.jenkins).getCredentials(Domain.global());
+        assertEquals(2, creds.size());
```

Comment:
Tests should be focused on a single behavior. Suggests separating this test into multiple discrete tests each testing a single behavior.

Reply:
tracking issue